README.md

```diff
@@ -1,30 +1,127 @@
 # WIP: Cerializer
-Cerializer aims at providing a even faster alternative to FastAvro and Avro standard library.
+Cerializer is an Avro de/serialization library that aims at providing a even faster alternative to FastAvro and Avro standard library.
 
-This speed increase does not come without a cost. Cerializer will work only with predefined set of schemata for which it will generate tailor made code. This way, the overhead caused by the universality of other serialization libraries will be avoided.
+This speed increase does not come without a cost. Cerializer will work only with predefined set of schemata for which it will generate tailor made Cython code. This way, the overhead caused by the universality of other serialization libraries will be avoided.
 
 Special credit needs to be given to [FastAvro](https://github.com/fastavro/fastavro) library, by which is this project heavily inspired.
 
+**Example of a schema and the corresponding code**
+
+SCHEMA
+```
+{
+    'name': 'array_schema', 
+    'doc': 'Array schema', 
+    'namespace': 'messaging', 
+    'type': 'record', 
+    'fields': [
+        {
+            'name': 'order_id', 
+            'doc': 'Id of order', 
+            'type': 'string'
+        }, 
+        {
+            'name': 'trades', 
+            'type': {
+                'type': 'array', 
+                'items': ['string', 'int']
+            }
+        }
+    ]
+}
+```
+
+CORRESPONDING CODE
+```
+def serialize(data, output):
```

Comment:
Won't this be faster if you use `cpdef` instead of `def`?

Reply:
Unfortunatelly, this is not possible.

`@cython.boundscheck(False)
@cython.wraparound(False)
@cython.initializedcheck(False)
def __invoke():
    cpdef serialize(data, output):
                                ^
------------------------------------------------------------

/Users/matejmicek/Library/Caches/cython/inline/_cython_inline_dcb32c39cfd21af4045a698eff759b64.pyx:15:33: C function definition not allowed here`